src/index.ts

```diff
@@ -1,17 +1,206 @@
+  export const rotateClockwise = 'image-editor:rotate-clockwise';
+  export const crop = 'image-editor:crop';
+  export const applyCrop = 'image-editor:apply-crop';
+  export const cancelCrop = 'image-editor:cancel-crop';
+}
+
+type IImageEditorTracker = IWidgetTracker<ImageEditorDocumentWidget>;
+export const IImageEditorTracker = new Token<IImageEditorTracker>('image-editor/tracki');
+
+const extension: JupyterFrontEndPlugin<IImageEditorTracker> = {
+    id: '@jupyterlab/image-editor-extension:plugin',
+    autoStart: true,
+    requires: [ILayoutRestorer, ICommandPalette],
+    provides: IImageEditorTracker,
+    activate
 };
 
-export default plugin;
+export default extension;
+
+function activate(
+    app: JupyterFrontEnd,
+    restorer: ILayoutRestorer,
+    palette: ICommandPalette,
+  ): IImageEditorTracker {
+  
+    const namespace = 'image-editor';
+    const tracker = new WidgetTracker<ImageEditorDocumentWidget>({ namespace });
+  
+    // Handle state restoration.
+    restorer.restore(tracker, {
+      command: 'docmanager:open',
+      args: widget => ({ path: widget.context.path, factory: FACTORY }),
+      name: widget => widget.context.path
+    });
+  
+    const factory = new ImageEditorFactory({
+      name: FACTORY,
+      modelName: 'base64',
+      fileTypes: ['png', 'jpg', 'jpeg'],
+      defaultFor: ['png', 'jpg', 'jpeg'],
+      commands: app.commands,
+      toolbarFactory: () => {
+        return [{
+          name: CommandIDs.rotateClockwise,
+          widget: new CommandToolbarButton(
+          {
+            commands: app.commands,
+            id: CommandIDs.rotateClockwise,
+            args: {
+              toolbar: true
+            }
+          })
+        },
+        {
+          name: CommandIDs.crop,
+          widget: new CommandToolbarButton(
+          {
+            commands: app.commands,
+            id: CommandIDs.crop,
+            args: {
+              toolbar: true
+            }
+          })
+        },
+        {
+          name: CommandIDs.applyCrop,
+          widget: new CommandToolbarButton(
+          {
+            commands: app.commands,
+            id: CommandIDs.applyCrop,
+            args: {
+              toolbar: true
+            }
+          })
+        },
+        {
+          name: CommandIDs.cancelCrop,
+          widget: new CommandToolbarButton(
+          {
+            commands: app.commands,
+            id: CommandIDs.cancelCrop,
+            args: {
+              toolbar: true
+            }
+          })
+        },
+        {
+          name: 'docmanager:save-as',
+          widget: new CommandToolbarButton(
+          {
+            commands: app.commands,
+            id: 'docmanager:save-as',
+            args: {
+              toolbar: true
+            }
+          })
+        }
+      ]
+      }
+    });
+  
+    factory.widgetCreated.connect((sender, widget) => {
+      widget.title.icon = 'jp-MaterialIcon jp-ImageIcon'; // TODO change
```

Comment:
You can get the proper icon with:
```suggestion
      const types = app.docRegistry.getFileTypesForPath(widget.context.path);

      if (types.length > 0) {
        widget.title.icon = types[0].icon!;
      }
```